src/createServer.js

```diff
@@ -1,3 +1,50 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const { createServer } = require('http');
+
+const { convertToCase } = require('./convertToCase');
+const { getErrorCollector } = require('./getErrorCollector');
+const { ERROR_MESSAGE, TO_CASE_QUERY } = require('./variables');
+
+module.exports.createServer = () =>
+  createServer((request, response) => {
+    response.setHeader('Content-Type', 'application/json');
+
+    const errorCollector = getErrorCollector();
+    const url = new URL(request.url, `http://${request.headers.host}`);
+    const originalText = url.pathname.slice(1);
+    const targetCase = url.searchParams.get(TO_CASE_QUERY);
+
+    if (!originalText) {
+      errorCollector.catchIfThrown(() => {
+        throw new Error(ERROR_MESSAGE.TEXT_REQUIRED);
```

Comment:
No need to throw new errors. Consider using a validation function as a middleware to check the response before the main logic. Or use a try-catch block here